Code/src/main/java/graph/Node.java

```diff
@@ -1,20 +1,26 @@
+package graph;
+
 import java.util.List;
 
-public class Node {
+public class Node<T> {
 	private int id;
+	private T value;
 	private List<Node> neighbours;
 	private int weightStorage;
 
-	public Node() {
+	public Node(T value) {
+		this.value = value;
```

Comment:
Nie wiem czy ten konstruktor nam się przyda. Właściwie jak tworzymy Node, to warto, żeby na starcie też już dostawał swoje ID